### ISSUE TITLE
System.InvalidOperationException in  Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.EventHookup.EventHookupCommandHandlerTests.EventHookupInUnformattedPosition1

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#2880]
While running tests on CI server:
http://dotnet-ci.cloudapp.net/job/dotnet_roslyn_windows_prtest/2766/console

```
 System.AggregateException : One or more errors occurred.\r\n---- System.AggregateException : One or more errors occurred.\r\n-------- System.AggregateException : One or more errors occurred.\r\n------------ System.AggregateException : One or more errors occurred.\r\n---------------- System.InvalidOperationException : Unexpected Null
        Stack Trace:
              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
              at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
           Src\Workspaces\Core\Portable\Workspace\Solution\DocumentState.cs(510,0): at Microsoft.CodeAnalysis.DocumentState.<GetSyntaxTreeAsync>d__31.MoveNext()
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
              at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
           Src\EditorFeatures\CSharp\EventHookup\EventHookupSessionManager_EventHookupSession.cs(172,0): at Microsoft.CodeAnalysis.Editor.CSharp.EventHookup.EventHookupSessionManager.EventHookupSession.<GetPlusEqualsTokenInsideAddAssignExpressionAsync>d__21.MoveNext()
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
              at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
           Src\EditorFeatures\CSharp\EventHookup\EventHookupSessionManager_EventHookupSession.cs(151,0): at Microsoft.CodeAnalysis.Editor.CSharp.EventHookup.EventHookupSessionManager.EventHookupSession.<DetermineIfEventHookupAndGetHandlerNameAsync>d__20.MoveNext()
           ----- Inner Stack Trace -----
              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
              at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
           Src\Workspaces\Core\Portable\Workspace\Solution\DocumentState.cs(147,0): at Microsoft.CodeAnalysis.DocumentState.<IncrementallyParseTreeAsync>d__11.MoveNext()
           ----- Inner Stack Trace -----
              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
              at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
           Src\Workspaces\Core\Portable\Workspace\Solution\DocumentState.cs(147,0): at Microsoft.CodeAnalysis.DocumentState.<IncrementallyParseTreeAsync>d__11.MoveNext()
           ----- Inner Stack Trace -----
              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
              at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
           Src\Workspaces\Core\Portable\Utilities\ValuesSources\RecoverableWeakValueSource.cs(105,0): at Microsoft.CodeAnalysis.Host.RecoverableWeakValueSource`1.<GetValueAsync>d__16.MoveNext()
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
              at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
           Src\Workspaces\Core\Portable\Workspace\Solution\RecoverableTextAndVersion.cs(99,0): at Microsoft.CodeAnalysis.RecoverableTextAndVersion.<GetValueAsync>d__12.MoveNext()
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
              at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
           Src\Workspaces\Core\Portable\Workspace\Solution\DocumentState.cs(106,0): at Microsoft.CodeAnalysis.DocumentState.<FullyParseTreeAsync>d__9.MoveNext()
           ----- Inner Stack Trace -----
           Src\Workspaces\Core\Portable\Utilities\Contract.cs(106,0): at Roslyn.Utilities.Contract.Fail(String message)
           Src\Workspaces\Core\Portable\Utilities\Contract.cs(73,0): at Roslyn.Utilities.Contract.ThrowIfNull[T](T value, String message)
           Src\Workspaces\Core\Portable\Workspace\Solution\RecoverableTextAndVersion.cs(126,0): at Microsoft.CodeAnalysis.RecoverableTextAndVersion.RecoverableText.<RecoverAsync>d__3.MoveNext()
```

**Comment by Pilchie** [comment]
@mattwar can you take a look?  @tmat do you know if we have a crash dump?

**Comment by tmat** [comment]
@Pilchie I don't think Jenkins saves crash dumps for test crashes, but I'm not positive. @jaredpar?

**Comment by jaredpar** [comment]
@tmat it does not. Even if it did though I don't see how it would have helped here because this didn't result in a cash dump, it's just a failed unit test.

**Comment by mattwar** [comment]
Smells like it could be related to #2832.

**Comment by mattwar** [comment]
This should be fixed by #2933
